### PR TITLE
feat(test): add `--logs` to `helm test`

### LINKED ISSUE #2923: Proposal: testing chart - show log of test pod
# Problem

Currently when we test our helm charts we get a Success or Fail. 

# Proposed solution
I propose to provide a flag that will allow the users to see the log of the test pod to get information about why it failed or passed

# Behavior 
In the `helm test <command>`  adding a flag`--log` or `--output` flag. In which, at the end of the test we output the test-pod log if flag is present. If the test pod is still present in the cluster(user did not use the clean up flag), the user should be able to get the output of the test-pod.

### PR DESCRIPTION
**Pull request by bacongobbler**
The original submitter hasn't responded in over a week, so I rebased #6167 off master and [addressed an earlier comment](https://github.com/helm/helm/pull/6167#discussion_r329748511).

supersedes #6167
closes #2923

### CODE REVIEW

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
So I keep getting this error when testing:
```
RUNNING: testingtest-nginx-service-test
UNKNOWN: testingtest-nginx-service-test: 
ERROR: error in opening log stream, got: container "curl" in pod "testingtest-nginx-service-test" is waiting to start: ContainerCreating
Error: 1 test(s) failed
```

This looks like it is not waiting for test completion

If I run it again, it finds the first pods logs, but looks like something is missing some formatting stuff:
```
RUNNING: testingtest-nginx-service-test
ERROR: pods "testingtest-nginx-service-test" already exists

POD LOGS: testingtest-nginx-service-test
  %!T(MISSING)otal    %!R(MISSING)eceived %!X(MISSING)ferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    36  100    36    0     0   1831      0 --:--:-- --:--:-- --:--:-- 36000
<h1>Hello</h1> <p>This is a test</p>
Error: 1 test(s) failed
```

**Review by thomastaylor312 — APPROVED** [review]
I paired with @bacongobbler to fix the issues on this PR

**Comment by bacongobbler** [comment]
@thomastaylor312 rightfully pointed out that we still need a second LGTM on this one before we merge. Thanks for keeping me honest. :sheep:

File: pkg/kube/client.go
Comment (thomastaylor312) [inline pkg/kube/client.go]: Fun story for the second person who reviews this: We were unintentionally selecting _all_ pods before this fix

File: pkg/releasetesting/environment.go
Comment (jdolitsky) [inline pkg/releasetesting/environment.go]: @bacongobbler everything lgtm - can you just explain/add a comment for this?

**Review by jdolitsky — APPROVED** [review]

File: pkg/releasetesting/environment.go
Comment (bacongobbler) [reply pkg/releasetesting/environment.go]: For sure!

**Pull request merged**
